python/tskit/tables.py

```diff
@@ -374,6 +374,43 @@ def __getitem__(self, index):
 
         return ret
 
+    def __setitem__(self, index, new_row):
+        """
+        Replaces a row of this table at the specified index with information from a
+        row-like object. Metadata, will be validated and encoded according to the table's
+        :attr:`metadata_schema<tskit.IndividualTable.metadata_schema>`.
+
+        :param index: the zero-index of the row to change
+        :param row-like new_row: An object that has attributes corresponding to the
+            properties of the new row. Both the objects returned from ``table[i]`` and
+            e.g. ``ts.individual(i)`` work for this purpose, along with any other
+            object with the correct attributes.
+        """
+        if isinstance(index, numbers.Integral):
+            # Single row by integer
+            if index < 0:
+                index += len(self)
+            if index < 0 or index >= len(self):
+                raise IndexError("Index out of bounds")
+        else:
+            raise TypeError("Index must be integer")
+
+        row_data = {
+            column: getattr(new_row, column)
+            for column in self.column_names
+            if "_offset" not in column
+        }
+
+        # Encode the meatdata - note that if this becomes a perf bottleneck it is
```

Comment:
"meatdata"

Reply:
Must have been hungry when writing this, fixed.